### PR TITLE
[BROOKLYN-153] Update the REST API documentation for the catalog endpoints

### PR DESCRIPTION
**Pull request by tbouron**

### CODE REVIEW

**Comment by sjcorbett** [comment]
Is "type" the right word for the documentation? It has too many meanings to be immediately understandable in this context. The methods the arguments are passed to use "symbolicName" (in brooklyn.catalog.BrooklynCatalog). Are either better than just "id"?

It's also worth changing the names of the parameters of the implementing methods in `CatalogResource` to match the interface.

**Comment by tbouron** [comment]
`type` is for me the best word here, that's why I used it. I didn't really understand `symbolicName` at first, not to mention that symbolicName represents something unique to me, like an ID. 

FYI, `symbolicName` is actually redundant as the API returns both `type` and `symbolicName` attributes (which have the same value every time)
 
That being said, I do agree that the `brooklyn.catalog.BrooklynCatalog` method's parameters need to reflect whatever name we choose for the API.

**Comment by neykov** [comment]
The user-visible term for catalog item is `id`. This is what the user sets in the catalog spec. The technical term is `symbolicName` with `type` kept for backwards compatibility with existing clients (it was removed initially but we had to put it back in).

**Comment by tbouron** [comment]
@neykov: Right, I was not aware of that. I guess I'm confused because we use two different terms internally and externally.

Anyway, is everyone ok if I change `type` by `symbolicName` then?

**Comment by neykov** [comment]
`symbolicName` makes sense as then the urls will become `.../{symbolicName}/{version}` which is consistent with the internal representation.

**Comment by tbouron** [comment]
Done. I also updated parameters' names within `CatalogResource` as @sjcorbett suggested.

**Comment by ahgittin** [comment]
`symbolicName` is the right unambiguous terminology -- good improvement

File: usage/rest-api/src/main/java/brooklyn/rest/api/CatalogApi.java [hunk 1]
```diff
@@ -180,15 +180,15 @@ public CatalogEntitySummary getEntity(
...
     @ApiErrors(value = {
         @ApiError(code = 404, reason = "Entity not found")
     })
     public CatalogEntitySummary getEntity(
-        @ApiParam(name = "entityId", value = "The ID of the entity or template to retrieve", required = true)
-        @PathParam("entityId") String entityId,
-        
+        @ApiParam(name = "symbolicName", value = "The symbolic name of the entity or template to retrieve", required = true)
+        @PathParam("symbolicName") String symbolicName,
+
```
Comment (sjcorbett) [inline usage/rest-api/src/main/java/brooklyn/rest/api/CatalogApi.java]: Surplus line

File: usage/rest-api/src/main/java/brooklyn/rest/api/CatalogApi.java [hunk 2]
```diff
@@ -202,18 +202,18 @@ public CatalogEntitySummary getEntity(
...
     @ApiErrors(value = {
         @ApiError(code = 404, reason = "Entity not found")
     })
     public CatalogEntitySummary getApplication(
-        @ApiParam(name = "applicationId", value = "The ID of the application to retrieve", required = true)
-        @PathParam("applicationId") String entityId,
-        
+        @ApiParam(name = "symbolicName", value = "The symbolic name of the application to retrieve", required = true)
+        @PathParam("symbolicName") String symbolicName,
+
```
Comment (sjcorbett) [inline usage/rest-api/src/main/java/brooklyn/rest/api/CatalogApi.java]: And here

Comment (sjcorbett) [inline usage/rest-api/src/main/java/brooklyn/rest/api/CatalogApi.java]: I guess you're just following the existing format? Doesn't matter, will merge.

Comment (tbouron) [inline usage/rest-api/src/main/java/brooklyn/rest/api/CatalogApi.java]: Strange, probably my Intellij not setup as it should be.

**Comment by neykov** [comment]
good to merge

**Comment by neykov** [comment]
@tbouron Could you close this.

**Comment by tbouron** [comment]
Done

**Pull request merged**